lib/ads/interstitial_ad_manager.js

```diff
@@ -502,21 +531,39 @@ shaka.ads.InterstitialAdManager = class {
       this.onEvent_(new shaka.util.FakeEvent(
           shaka.ads.Utils.AD_SKIP_STATE_CHANGED));
     }
+    const eventsSent = [];
     this.adEventManager_.listenOnce(this.player_, 'error', error);
-    this.adEventManager_.listenOnce(this.player_, 'firstquartile', () => {
-      updateBaseVideoTime();
-      this.onEvent_(
-          new shaka.util.FakeEvent(shaka.ads.Utils.AD_FIRST_QUARTILE));
-    });
-    this.adEventManager_.listenOnce(this.player_, 'midpoint', () => {
-      updateBaseVideoTime();
-      this.onEvent_(
-          new shaka.util.FakeEvent(shaka.ads.Utils.AD_MIDPOINT));
-    });
-    this.adEventManager_.listenOnce(this.player_, 'thirdquartile', () => {
-      updateBaseVideoTime();
-      this.onEvent_(
-          new shaka.util.FakeEvent(shaka.ads.Utils.AD_THIRD_QUARTILE));
+    this.adEventManager_.listen(this.video_, 'timeupdate', () => {
+      const duration = this.video_.duration;
+      if (!duration) {
+        return;
+      }
+      if (interstitial.isSkippable && interstitial.skipOffset &&
+          ad.canSkipNow() && ad.getRemainingTime() > 0 &&
+          ad.getDuration() > 0) {
+        this.onEvent_(
+            new shaka.util.FakeEvent(shaka.ads.Utils.AD_SKIP_STATE_CHANGED));
+      }
+      const currentTime = this.video_.currentTime;
```

Comment:
The if-then chain below could be shorter if this was expressed as a percent or ratio.

Like, if this was:

```js
const currentPercent = 100 * this.video_.currentTime / duration;
```

Then below you could just do (if you are also making `eventsSent` into a set):

````js
if (currentPercent >= 25 && !eventsSent.has('firstquartile')) {
````

Which can probably fit on one line.

Reply:
Done!